Web Inspector: No search results

francescorn opened the pull request:
#### 6d291b6825cc243898c27d8d97d35a20a2ca0b64
<pre>
Web Inspector: No search results
<a href="https://bugs.webkit.org/show_bug.cgi?id=259843">https://bugs.webkit.org/show_bug.cgi?id=259843</a>

Reviewed by Devin Rousso.

Web Inspector search is not working due to _inputFieldSearch(event) not
being triggered.

Replace incremental with throttle and replace &quot;search&quot; with &quot;input&quot;.

Import Throttler class to add _inputFieldInputThrottler function to handle throttler for &quot;input&quot; using fire function.
Incorporate 250 ms delay.

In _inputFieldKeyDown use findBannerRevealPreviousResult and findBannerRevealNextResult for shift and enter key presses
replacing _searchBackwards and _searchForwards.

Delete _inputFieldKeyUp.

Delete _searchBackwards and _searchForwards in _inputFieldInput(event).

* Source/WebInspectorUI/UserInterface/Views/FindBanner.js:
(WI.FindBanner.prototype._inputFieldKeyDown):
(WI.FindBanner.prototype._inputFieldInput):
(WI.FindBanner.prototype._inputFieldKeyUp): Deleted.
(WI.FindBanner.prototype._inputFieldSearch): Deleted.

Canonical link: <a href="https://commits.webkit.org/266783@main">https://commits.webkit.org/266783@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/93dba04455df28b7f059b8a1a9ce3f2b792011dc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/14742 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/15049 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/15401 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/16490 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/13898 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/17570 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/15148 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/16553 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/14922 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/15424 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/12516 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/17225 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/12708 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/13299 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/20293 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/13782 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/13466 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/16707 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/14011 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/11852 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/13305 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3556 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/17642 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/13854 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dcrousso reviewed (changes requested):
thanks for taking the time to investigate and propose a fix! 😃

this used to work, so id prefer we figure out what caused it to break before doing any fixes

it looks like the `"search"` event (and `incremental` attribute) were both removed in [266068@main](https://commits.webkit.org/266068@main) (#15788).  we probably need to go through and replace all `"search"` event listeners with `"input"`/`"change"` (depending on whether it was `incremental` or not) event listeners

your solution is arguably on the right track in that the event listeners that used to work aren't anymore, but i think a better approach would be to fix that root cause rather than manually calling the event listener elsewhere 🙂

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/FindBanner.js:
```diff
@@ -40,16 +40,21 @@ WI.FindBanner = class FindBanner extends WI.NavigationItem
         this._resultCountLabel = document.createElement("label");
         this.element.appendChild(this._resultCountLabel);
 
+        this._throttler = new Throttler(this._inputFieldInput.bind(this), 150);
```
style: this needs a better name like `_inputFieldInputThrottler`

nit: we normally do timeouts of either `100` or `250`.  in this case id probably go with `250`

francescorn replied:
Added this change in pr:
```
this._inputFieldInputThrottler = new Throttler(this._inputFieldInput.bind(this), 250);
```

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/FindBanner.js:
```diff
@@ -40,16 +40,21 @@ WI.FindBanner = class FindBanner extends WI.NavigationItem
 
+        this._throttler = new Throttler(this._inputFieldInput.bind(this), 150);
+
+        this._handleThrottlerInput = function() {
```
rather than save this to a member variable id just inline it
```
this._inputField.addEventListener("input", (event) => {
    this._inputFieldInputThrottler.fire();
}, false);
```

francescorn replied:
Added this change in pr.

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/FindBanner.js:
```diff
@@ -231,8 +236,10 @@ WI.FindBanner = class FindBanner extends WI.NavigationItem
             this._searchBackwards = true;
-        else if (event.keyIdentifier === "Enter")
+        else if (event.keyIdentifier === "Enter") {
             this._searchKeyPressed = true;
```
i dont actually think we need `_searchKeyPressed` anymore now that we're manually calling `_inputFieldInput`.  i think it was only used previously since `"search"` would be dispatched separately from `"keydown"` and we needed a way of tracking state between them.  id recommend moving the code inside `_inputFieldInput` that is gated by `_searchKeyPressed` into this function instead (and removing all mentions of `_searchKeyPressed` and `_searchBackwards ` since they're not really useful anymore)
```
else if (event.keyIdentifier === "Enter") {
    if (this._numberOfResults > 0) {
        if (event.shiftKey)
            this._delegate?.findBannerRevealPreviousResult?.(this);
        else
            this._delegate?.findBannerRevealNextResult?.(this);
    }
}
```

francescorn replied:
Added this change in pr:
```
if (event.keyIdentifier === "Enter") {
     if (this._numberOfResults > 0) {
           if (event.shiftKey) 
                 this._delegate?.findBannerRevealPreviousResult?.(this);
           else 
                 this._delegate?.findBannerRevealNextResult?.(this);
     }
}
```
Wondering if possibly creating an event property enterKey to the enter key press rather than using a keyIdentifier.

dcrousso reviewed (changes requested):
this is great work!

i think we should also do similar fixes for `WI.FilterBar` and `WI.SearchSidebarPanel` since they both use `<input type="search">` too.  im fine with those being separate PRs or if you wanted to add fixes for those to this PR too 

a few structural/style comments, but otherwise it looks pretty good!  i think one more iteration and we should be good to go 😃

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/FindBanner.js:
```diff
@@ -227,14 +229,17 @@ WI.FindBanner = class FindBanner extends WI.NavigationItem
-    }
+    _inputFieldKeyDown(event) {
+        if (event.keyIdentifier === "Enter") {
+            if (this._numberOfResults > 0) {
```
please remove this similar code from `_inputFieldInput` as it's not really reachable/needed there anymore

francescorn replied:
Removed in pr.

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/FindBanner.js:
```diff
@@ -227,14 +229,17 @@ WI.FindBanner = class FindBanner extends WI.NavigationItem
+            }
+        }
+    }
     _inputFieldKeyUp(event)
```
i think we dont need this or the `"keyup"` anymore

francescorn replied:
Removed in pr.

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/FindBanner.js:
```diff
@@ -227,14 +229,17 @@ WI.FindBanner = class FindBanner extends WI.NavigationItem
-    _inputFieldKeyDown(event)
-    {
-        if (event.keyIdentifier === "Shift")
-            this._searchBackwards = true;
```
please remove all mentions of `_searchBackwards` from this file now that it's not used/needed anymore

francescorn replied:
Removed in pr.

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/FindBanner.js:
```diff
@@ -227,14 +229,17 @@ WI.FindBanner = class FindBanner extends WI.NavigationItem
-        if (event.keyIdentifier === "Shift")
-            this._searchBackwards = true;
-        else if (event.keyIdentifier === "Enter")
-            this._searchKeyPressed = true;
```
ditto above for `_searchKeyPressed`

francescorn replied:
Removed in pr.

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/FindBanner.js:
```diff
@@ -227,14 +229,17 @@ WI.FindBanner = class FindBanner extends WI.NavigationItem
-        else if (event.keyIdentifier === "Enter")
-            this._searchKeyPressed = true;
-    }
+    _inputFieldKeyDown(event) {
```
style: the `{` should go on its own line (like it was before)

francescorn replied:
Fixed in pr.

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/FindBanner.js:
```diff
@@ -44,12 +44,14 @@ WI.FindBanner = class FindBanner extends WI.NavigationItem
-        this._inputField.addEventListener("search", this._inputFieldSearch.bind(this), false);
+        this._inputFieldInputThrottler = new Throttler(this._inputFieldInput.bind(this), 250);
+        this._inputField.addEventListener("input", (event) => {
+            this._inputFieldInputThrottler.fire();
```
we should pass along the most recent `event` just in case `_inputFieldInput` needs to use it
```
this._inputFieldInputThrottler.fire(event);
```

francescorn replied:
Fixed in pr.

dcrousso reviewed (approved):
nice work!  thanks for iterating with me :)

francescorn committed:
Web Inspector: No search results
https://bugs.webkit.org/show_bug.cgi?id=259843

Reviewed by Devin Rousso.

Web Inspector search is not working due to _inputFieldSearch(event) not
being triggered.

Replace incremental with throttle and replace "search" with "input".

Import Throttler class to add _inputFieldInputThrottler function to handle throttler for "input" using fire function.
Incorporate 250 ms delay.

In _inputFieldKeyDown use findBannerRevealPreviousResult and findBannerRevealNextResult for shift and enter key presses
replacing _searchBackwards and _searchForwards.

Delete _inputFieldKeyUp.

Delete _searchBackwards and _searchForwards in _inputFieldInput(event).

* Source/WebInspectorUI/UserInterface/Views/FindBanner.js:
(WI.FindBanner.prototype._inputFieldKeyDown):
(WI.FindBanner.prototype._inputFieldInput):
(WI.FindBanner.prototype._inputFieldKeyUp): Deleted.
(WI.FindBanner.prototype._inputFieldSearch): Deleted.

Canonical link: https://commits.webkit.org/266783@main
